src/shared/api/dto.ts

```diff
@@ -1,183 +1,228 @@
 
-  interface PaginationParams {
-    page: number
-    page_size: number
-  }
+const metadataSchema = z.record(z.string(), z.any());
+
+const accountIdSchema = z.string().brand('AccountId');
+const domainIdSchema = z.string().brand('DomainId');
 
-  interface Paginated<T> {
-    pagination: {
-      page: number
-      page_size: number
-      total: number
-    }
-    data: T[]
+class AssetDefinitionId {
+  private assetName = '';
+  private assetDomain = '';
+
+  public constructor(name: string, domain: string) {
+    this.assetName = name;
+    this.assetDomain = domain;
   }
 
-  interface Asset {
-    account_id: string
-    definition_id: string
-    value: AssetValue
+  public get name() {
+    return this.assetName;
+  }
+  public get domain() {
+    return this.assetDomain;
+  }
+  public toString() {
+    return this.assetName + '#' + this.assetDomain;
   }
+}
 
-  type AssetValue =
-    | Tagged<'Quantity', string>
-    | Tagged<'BigQuantity', string>
-    | Tagged<'Fixed', string>
-    | Tagged<'Store', JsonObject>;
+export function transformToAssetDefinitionId(assetDefinitionId: string): AssetDefinitionId {
+  const parts = assetDefinitionId.split('#');
+  invariant(parts.length === 2, 'Invalid assetDefinitionId format');
 
-  type AssetValueType = 'Quantity' | 'BigQuantity' | 'Fixed' | 'Store';
+  const [asset, asset_domain] = parts;
 
-  interface AssetDefinition {
-    id: string
-    value_type: AssetValueType
-    mintable: Mintable
-  }
+  return new AssetDefinitionId(asset, asset_domain);
+}
 
-  type Mintable = 'Once' | 'Infinitely' | 'Not';
+class AssetId {
+  private assetName = '';
+  private assetDomain = '';
+  private accountSignatory = '';
+  private accountDomain = '';
 
-  interface Peer {
-    address: string
-    public_key: PublicKey
+  public constructor(data: { asset: string, asset_domain: string, signatory: string, account_domain: string }) {
+    this.assetName = data.asset;
+    this.assetDomain = data.asset_domain;
+    this.accountSignatory = data.signatory;
+    this.accountDomain = data.account_domain;
   }
 
-  interface Role {
-    id: string
-    permissions: PermissionToken[]
+  public get asset() {
+    return this.assetName;
   }
-
-  interface PermissionToken {
-    name: string
-    params: any
+  public get domain() {
+    return this.assetDomain ? this.assetDomain : this.accountDomain;
   }
-
-  interface Account {
-    id: string
-    assets: Asset[]
-    signatories?: PublicKey[]
-    permission_tokens?: PermissionToken[]
-    roles: Role[]
-    signature_check_condition?: any
-    metadata: any
+  public get signatory() {
+    return this.accountSignatory;
   }
-
-  interface Domain {
-    id: string
-    accounts: Account[]
-    asset_definitions: AssetDefinition[]
-    logo: null | string
-    metadata: any
-    /**
-     * amount of triggers, always 0 for now
-     */
-    triggers: number
+  public get account_domain() {
```

Comment:
We don't have ESLint being setup properly, I guess, but it is common across the codebase to use `camelCase`.

```suggestion
  public get accountDomain() {
```